libminifi/test/unit/MemoryUsageTest.cpp

```diff
@@ -24,14 +25,16 @@
 TEST_CASE("Test Physical memory usage", "[testphysicalmemoryusage]") {
   constexpr bool cout_enabled = true;
   std::vector<char> v(30000000);
+  // use v to prevent recent compilers from optimizing out the allocation
```

Comment:
It likely has something to do with `std::vector` being constexpr-ready in C++20 and I assume the optimizer executes such code at compile-time regardless of `constexpr` at the usage site. Printing the contents to the console keeps this code working for now, but there is nothing preventing future compilers from seeing through the trick here.